File: src/main/java/com/survivalMode/Highscore/Models/Score.java
package com.survivalMode.Highscore.Models;

import lombok.Getter;
import java.util.Objects;

@Getter
public final class Score {
    public final String username;
    private final int combatLevel;
    private final int totalLevel;
    private final int totalTimePlayed;

    public Score(String user, int combatLevel, int totalLevel, int timePlayed) {
        this.username = user;
        this.combatLevel = combatLevel;
        this.totalLevel = totalLevel;
        this.totalTimePlayed = timePlayed;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (!(o instanceof Score)) return false;
        Score score = (Score) o;
        return username.equals(score.username);
    }

    @Override
    public int hashCode() {
        return Objects.hash(username);
    }

    @Override
    public String toString() {
        return "Username: "   + username +
           ", Combat Level: " + combatLevel +
           ", Total Level: "  + totalLevel;
    }
}

File: src/main/java/com/survivalMode/services/highscore/models/Score.kt
package com.survivalMode.services.highscore.models

import lombok.Getter
import java.util.*

@Getter
class Score(
    private val username: String,
    private val combatLevel: Int,
    private val totalLevel: Int,
    private val totalTimePlayed: Int
) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Score) return false
        return username == other.username
    }

    override fun hashCode(): Int {
        return Objects.hash(username)
    }

    override fun toString(): String {
        return "Username: " + username +
                ", Combat Level: " + combatLevel +
                ", Total Level: " + totalLevel +
                ", Time Played: " + totalTimePlayed
    }
}
